Data reformatting - 1

trentwirth opened the pull request:
first stage of data reformatting
- wrote code with Jon and Aaron that grabs the correct qualisys marker labels out of `.tsv` turned `dict`
- wrote code solo that checks for a saved `.json` file that holds the qualisys dictionary, and if it exists, it loads it rather than recreating it every iteration. This makes the code feel 10x faster.

jonmatthis commented on main.py:
```diff
@@ -0,0 +1,47 @@
+    qualisys_df = pd.read_csv(filepath_or_buffer=str(qualisys_data_path), delimiter='\t', header=11)
+    pupil_df = pd.read_csv(filepath_or_buffer=str(pupil_data_path), delimiter=',', header=0)
+
+    if exists('utilities/qualisys_dict.json'):  # check and see if I've run this code before/ created the dictionary
```
1. NO PATHS AS STRINGS!!!!!!!!!!!

 Use `pathlib` to create a path to where the file should be and then check if it exists with `path_to_json.is_file()` or `...exists()` (or something like that)
 
2. the `json` is  ✨ data ✨  not code- don't save it inside of this repo!! Save it somewhere in the subject's data folder

jonmatthis reviewed (changes requested):
There's enough stuff in your `main.py` that you should probably abstract them all into functions or classes

New goal for the primary architecture - Your `main.py` file consists of nothing but a list of subject ID's and a for-loop that calls a function called `process_data` or something that does all the work. 

The goal is that by the time the code gets to the end of `main`, there are no/minimal superfluous variables laying around. 

Bonus points - Set it up so that you never need to alter `main.py` again after this PR (other than updating the list of subject ID's)

trentwirth commented:
@jonmatthis I updated the code and re-requested review for the two commits `created process data and removed datafiles from repo` and `Update process_data.py` 

I'm not sure if re-requesting was the right move here

jonmatthis commented:
Yeah, that's as good a way as any to cue me with "please look at this again"

jonmatthis commented on main.py:
```diff
@@ -0,0 +1,34 @@
+    qualisys_df = pd.read_csv(filepath_or_buffer=str(qualisys_data_path), delimiter='\t', header=11)
+    pupil_df = pd.read_csv(filepath_or_buffer=str(pupil_data_path), delimiter=',', header=0)
+
+    subject_json_path = qualisys_data_path = base_data_path / subject_id / 'processing_jsons'
```
I'm surprised it's even possible to do this `thing = thing = thing` thing 😂 

break this up into 2 lines plz

trentwirth replied:
oo, this was a typo,  that second "=" should be a "/"

jonmatthis commented on main.py:
```diff
@@ -0,0 +1,34 @@
+
+    subject_json_path = qualisys_data_path = base_data_path / subject_id / 'processing_jsons'
+
+    process_data(subject_json_path, qualisys_df, pupil_df)
```
Sending path down from `main` is correct! 👍 

Now if you want to change your folder structure, you just update this `main` and everything downstream stays the same

jonmatthis commented on proccess_data.py:
```diff
@@ -0,0 +1,18 @@
+                 qualisys_df: pandas.core.frame.DataFrame,
+                 pupil_df: pandas.core.frame.DataFrame) -> np.array:
+
+    subject_qualisys_json_path = subject_json_path / 'qualisys_dict.json'
```
I think you can do this in main too

jonmatthis commented on requirements.txt:
```diff
@@ -0,0 +1,2 @@
+numpy==1.23.2
+git+git://github.com/HuMoN-Research-Lab/shared_utilities
```
update requirements - use `pip list` to see what you have installed in your env

jonmatthis commented on utilities/qualisys_json_creator.py:
```diff
@@ -0,0 +1,26 @@
+import pandas
+
+
+def qualisys_json_creator(subject_qualisys_json_path: Path,
```
Nice! ✨

jonmatthis reviewed (changes requested):
Just a few little things, then good to go!

trentwirth commented:
I accidentally responded to these changes on a clean branch, so I just merged that branch into the `Data Reformatting` branch we've been working on. The notes are above, but I'm copying them below for posterity:

moved *all* path control into `main.py` and updated `requirements.txt` - I put everything that was in my conda environment in that `requirements.txt` file, I'm not sure if I need everything in there but it probably doesn't hurt...

also got rid of the weird double "=" from line 14 of `process_data.py`. It was clearly a copy/paste mistake, and I too an surprised pycharm was like "this is fine"

jonmatthis commented on requirements.txt:
```diff
@@ -0,0 +1,33 @@
+numpy==1.23.2
```
This isn't the worst thing in the world, but I think its cleaner if you only include the things you would have `pip installed`'d if you weren't using this `requirements.txt` file. 

In your case, I think thatd be: 

numpy
matplotlib (even though you haven't used it yet...)
pandas

(plus their version numbers)

jonmatthis commented:
Right on - 

In the future - it's better to to keep all conversation about specific parts of the code in the comments attached to those lines (i.e. don't reply to the comment and then leave ANOTHER comment talking that references that conversation)

>  I too an surprised pycharm was like "this is fine"

Important brain thing -> The correct way to say this is that you were surprised that ***_python_*** allowed that syntax to work NOT 'pycharm' 

Python is the language that defines the syntax - PyCharm is just the IDE. What you wrote is like confusing "MS Word" with "The Rules of English Grammar" :D

trentwirth commented:
I am both surprised that python allows something like that, and that pycharm doesn't flag it as "not great code" 😅
